akvo/rsr/management/commands/audit_program_aggregation.py

```diff
@@ -0,0 +1,52 @@
+from django.core.management.base import BaseCommand
+from akvo.rsr.usecases import period_update_aggregation as aggregation
+from django.db.models import Q
+from akvo.rsr.models import ProjectHierarchy, IndicatorPeriod
+from akvo.rsr.models.result.utils import QUANTITATIVE, PERCENTAGE_MEASURE, calculate_percentage
+from akvo.utils import ensure_decimal
+
+
+def audit_period_aggregation(period):
+    count = 0
+    value, numerator, denominator = aggregation.sum_updates(period)
+    if period.indicator.measure == PERCENTAGE_MEASURE:
+        contrib_numerator, contrib_denominator = aggregation.sum_contributed_percentage_value(period)
+        numerator = ensure_decimal(numerator) + ensure_decimal(contrib_numerator)
+        denominator = ensure_decimal(denominator) + ensure_decimal(contrib_denominator)
+        value = calculate_percentage(numerator, denominator)
+    else:
+        value = ensure_decimal(value) + aggregation.sum_contributed_unit_value(period)
+
+    try:
+        assert ensure_decimal(period.actual_value) == value
+    except AssertionError:
+        print('AssertionError', period.actual_value, value, period.id, str(period), period.indicator.result.project.id, str(period.indicator.result.project))
+    count += 1
+
+    for child_period in period.child_periods.all():
+        count += audit_period_aggregation(child_period)
+
+    return count
+
+
+class Command(BaseCommand):
+
+    def add_arguments(self, parser):
+        parser.add_argument('program_id', type=int)
+
+    def handle(self, *args, **options):
+        try:
+            hierarchy = ProjectHierarchy.objects.get(root_project=options['program_id'])
+            program = hierarchy.root_project
+        except ProjectHierarchy.DoesNotExist:
+            print("Program not found")
+            return
+
+        running_quantitative_periods = IndicatorPeriod.objects\
+            .filter(indicator__result__project=program, indicator__type=QUANTITATIVE)\
+            .exclude(Q(actual_value__isnull=True) | Q(actual_value__exact=''))
+
+        count = 0
+        for period in running_quantitative_periods:
+            count += audit_period_aggregation(period)
```

Comment:
I think it would be good to put this into a usecase module with a test. Otherwise when we merge this into `master`, it might break and we won't notice it. (I can't remember if `ProjectHierarchy` still exists as a model)

Reply:
`ProjectHierarchy` is a model used to indicate that a project is a program. It's not a proper name but that's what we have.

I've updated the code to accept `project_id` as a parameter so it's not limited to just programs.